docs/reference/search/search-your-data/learning-to-rank.asciidoc

```diff
@@ -0,0 +1,109 @@
+[[learning-to-rank]]
+== Learning To Rank
+
+preview::["The Learning To Rank feature is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but this feature is not subject to the support SLA of official GA features."]
+
+NOTE: This feature is available for Elastic Stack versions 8.12.0 and newer and requires a https://www.elastic.co/pricing[Platinum subscription] or higher.
+
+Learning To Rank (LTR) is a technique used to build a complex machine learning (ML) -based ranking function for your search engine.
+
+The LTR function takes a list of documents and a search context and outputs ranked documents:
+
+[[learning-to-rank-overview-diagram]]
+.Learning To Rank overview
+image::images/search/learning-to-rank-overview.png[Learning To Rank overview,align="center"]
+
+
+[discrete]
+[[learning-to-rank-search-context]]
+=== Search context
+
+In addition to the list of documents to sort, the LTR function also requires a search context. Typically, this search context includes at least the search terms provided by the user ('text_query' in the example above).
+
+The search context can also be a lot more complex and provides information like demographic data (geolocation, age, …) data for the user who submitted the search.
```

Comment:
Also, I'd recommend adding some additional examples that aren't user-specific, e.g.

"The search context can also provide additional information used in the ranking model. This could be information about the user doing the searching (like demographic data, geolocation, or age); about the query (like query length); or document in the context of the query (like score for the title field)."

Reply:
New version is a mix of various suggestions